site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,181 @@
+# Hosting: Firebase Functions
+
+This tutorial will guide you through the process of deploying your bot to [Firebase Functions](https://firebase.google.com/docs/functions).
+
+## Prerequisites
+
+To follow along, you will need to have a Google account.
+If you don't already have one, you can create it [here](https://accounts.google.com/signup).
+
+## Setup
+
+This section guides you through the setup process.
+If you need more detailed explanations on each step that you'll take, refer to the [official Firebase documentation](https://firebase.google.com/docs/functions/get-started).
+
+### Creating a Firebase Project
+
+1. Go to the [Firebase console](https://console.firebase.google.com/) and click **Add Project**.
+2. If prompted, review and accept the Firebase terms.
+3. Click **Continue**.
+4. Decide on whether you want to share analytics or not.
+5. Click **Create Project**.
+
+### Setting Things Up
+
+To write functions and deploy them to the Firebase Functions runtime, you'll need to set up a Node.js environment and install the Firebase CLI.
+
+> It's important to note that only the Node.js versions 14, 16, and 18 are currently supported by Firebase Functions.
+> For more on the supported Node.js versions, refer to [here](https://firebase.google.com/docs/functions/manage-functions#set_nodejs_version).
+
+Once you have Node.js and NPM installed, install Firebase CLI globally:
+
+```sh
+npm install -g firebase-tools
+```
+
+### Initializing the Project
+
+1. Run `firebase login` to open your browser and authenticate Firebase CLI with your account.
+2. `cd` into your project's directory.
+3. Run `firebase init functions`, and type `y` when asked whether you want to initialize a new codebase.
+4. Choose `use existing project` and select the project you created in Step 1.
+5. The CLI gives you two options for language support:
+   - JavaScript
+   - TypeScript
+6. Optionally, you can select ESLint.
+7. The CLI asks you if you want to install the dependencies with npm.
+   If you use another package manager like `yarn` or `pnpm` you can decline.
+   In that case, you have to `cd` into the `functions` directory and install the dependencies manually.
+8. Open `./functions/package.json` and look for the key: `"engines": {"node": "16"}`.
+   The `node` version should match your installed version of Node.js.
+   Otherwise, the project might not run.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point:
+
+```ts
+import * as functions from "firebase-functions";
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot(""); // <-- put your bot token here (https://t.me/BotFather)
```

Comment:
```suggestion
const bot = new Bot(""); // <-- put your bot token here
```

Reply:
If we remove the link to where people can get their token, then we should add a paragraph the describes this. Should I add that?

Reply:
No, don’t.

Reply:
Let's merge this as-is and take care of it in #664 then

Reply:
```suggestion
const bot = new Bot(""); // <-- put your bot token between the ""
```